### PR TITLE
Setting Node to current stable (0.12) on TravisCI

### PR DESCRIPTION
**Pull request by mistakevin**
Also remove pin for binstar...

### CODE REVIEW

**Comment by mistakevin** [comment]
It turns out that TravisCI already utilizes `nvm`, and allows us to call it from within `.travis.yml`. This PR will set it to pull in the latest stable release of node. You can see [here](http://docs.travis-ci.com/user/build-environment-updates/2015-04-09/) that travis tried to set the default node to 0.12.*, but it was causing problems with their tests so they retracted it. @damianavila @bryevdv

File: scripts/travis_install [hunk 1]
```diff
@@ -14,13 +14,13 @@ rm -rf $MINICONDA.sh
...
-PINNED_PKGS=$(cat <<EOF
-binstar ==0.10.1
-EOF
-)
-echo -e "$PINNED_PKGS" > /home/travis/miniconda/conda-meta/pinned
+# PINNED_PKGS=$(cat <<EOF
+# binstar ==0.10.1
+# EOF
+# )
+# echo -e "$PINNED_PKGS" > /home/travis/miniconda/conda-meta/pinned
```
Comment (bryevdv) [inline scripts/travis_install]: Will this work uncommented, but with just deleting the binstar line? I think that would be preferable from a diff perspective.

Comment (mistakevin) [inline scripts/travis_install]: I agree, Aaron and I were just testing that it would work using anaconda. It might even want to go into another PR unless @damianavila is ok with keeping it in here.

File: scripts/travis_install [hunk 2]
```diff
@@ -15,12 +15,12 @@ rm -rf $MINICONDA.sh
...
 
 PINNED_PKGS=$(cat <<EOF
-binstar ==0.10.1
+
 EOF
 )
 echo -e "$PINNED_PKGS" > /home/travis/miniconda/conda-meta/pinned
 
-DEPS_TRAVIS="python=$TRAVIS_PYTHON_VERSION conda-build jinja2"
+DEPS_TRAVIS="python=$TRAVIS_PYTHON_VERSION conda-build jinja2 anaconda-client"
```
Comment (damianavila) [inline scripts/travis_install]: `anaconda-client` will be used when we upload packages. in the past, I installed `binstar` here: https://github.com/bokeh/bokeh/blob/master/.travis.yml#L55
Let's put the anaconda-client in the same place... they are "upload" dependencies, so I would like to have it separated... what do you think?

**Comment by damianavila** [comment]
LGTM, I will merge it tomorrow unless there is someone else asking for more time to review...

**Comment by bryevdv** [comment]
LGTM (inspected all test output)
